packages/rum-vue/src/route-hooks.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+export function routeHooks(router, apm) {
+  let transaction
+  router.beforeResolve((to, from, next) => {
+    /**
+     * Get the last matched route record which acts as stack when
+     * route changes are pushed and popped out of the stack
+     */
+    const matched = to.matched || []
+    let path = to.path
+    if (matched.length > 0) {
+      path = matched[matched.length - 1].path || path
+    }
+    transaction = apm.startTransaction(path, 'route-change', {
+      managed: true,
+      canReuse: true
+    })
+    next()
+  })
+
+  router.afterEach(() => {
```

Comment:
We can also use `beforeRouteEnter` and pass a callback to the `next` function which will be called after the DOM is updated. See [the 12th step in navigation flow](https://router.vuejs.org/guide/advanced/navigation-guards.html#in-component-guards)

Reply:
It's a component level hook and wont work for us, We don't expose any component in our package.